### ISSUE TITLE
MODX 3 Lack of access when working with a media source

### ISSUE THREAD

**Issue by Ruslan-Aleev** [modxcms/revolution#14714]
### Summary
I created a new file source and linked it to the assets/ folder.

When I try to upload a file to TV, or open a file source, an access error appears:

![source](https://user-images.githubusercontent.com/12523676/64184879-3829ad00-ce7d-11e9-8ff8-3c836f368570.gif)

### Environment
MODX 3

**Comment by Mark-H** [comment]
Can't reproduce; see my configuration:

![Schermafbeelding 2019-09-03 om 17 07 40](https://user-images.githubusercontent.com/312944/64185437-6dc69a00-ce6d-11e9-9729-dc084a8e2606.jpg)
![Schermafbeelding 2019-09-03 om 17 08 05](https://user-images.githubusercontent.com/312944/64185445-715a2100-ce6d-11e9-8aaa-8667142df903.jpg)
![Schermafbeelding 2019-09-03 om 17 07 52](https://user-images.githubusercontent.com/312944/64185459-761ed500-ce6d-11e9-8787-44db5049cbba.jpg)

**Comment by Ruslan-Aleev** [comment]
Yes, everything is the same with me, although in the "Source Type" it is different for me - File System.

So I tried to manually add the user:
![source2](https://user-images.githubusercontent.com/12523676/64186185-7627d080-ce7f-11e9-916e-75899ad937aa.gif)

**Comment by Ruslan-Aleev** [comment]
Error log:
```
[03-Sep-2019 18:09:41 Europe/Moscow] PHP Fatal error:  Uncaught Error: Call to a member function hasPermission() on array in /home/a/aleeviz8/m3/public_html/core/cache/mgr/smarty/default/89692df6532ffacaf14a4abf11b3ba3c3664444a_0.file.onlineusers.tpl.php:86
Stack trace:
#0 /home/a/aleeviz8/m3/public_html/core/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_5d6e82354033d3_17110516(Object(Smarty_Internal_Template))
#1 /home/a/aleeviz8/m3/public_html/core/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#2 /home/a/aleeviz8/m3/public_html/core/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(216): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#3 /home/a/aleeviz8/m3/public_html/core/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(232): Smarty_Internal_Template->render(false, 0)
#4 /home/a/aleeviz8/m3/public_html/core/vendor/smarty/smarty/libs/sysplugins/smarty_in in /home/a/aleeviz8/m3/public_html/core/cache/mgr/smarty/default/89692df6532ffacaf14a4abf11b3ba3c3664444a_0.file.onlineusers.tpl.php on line 86
[03-Sep-2019 18:14:04 Europe/Moscow] PHP Fatal error:  Class 'modUserGroupRoleGetListProcessor' not found in /home/a/aleeviz8/m3/public_html/core/src/Revolution/Processors/Security/Role/GetAuthorityList.php on line 28
```

**Line 86 in 89692df6532ffacaf14a4abf11b3ba3c3664444a_0.file.onlineusers.tpl.php:**
`<?php if ($_smarty_tpl->tpl_vars['can_view_logs']->value) {?>`

**Line 28 in GetAuthorityList.php:**
`class GetAuthorityList extends modUserGroupRoleGetListProcessor`

**Comment by Mark-H** [comment]
That hasPermission error is fixed in https://github.com/modxcms/revolution/pull/14710 (merged today) and the role dropdown not working in https://github.com/modxcms/revolution/pull/14706 (not yet merged)

**Comment by Ruslan-Aleev** [comment]
Yes, I have the current version of MODX 3:
Fix permission check causing fatal error on dashboard [#14707] (#14710)

I tried with PR https://github.com/modxcms/revolution/pull/14706, but there is also an error.

**Comment by Ruslan-Aleev** [comment]
@JoshuaLuckers This problem may be related with https://github.com/modxcms/revolution/pull/14704 similarly https://github.com/modxcms/revolution/issues/14687#issuecomment-527532658 ?

**Comment by petyagrill** [comment]
I tried to reproduce, there are no errors. Everything works as expected.

**Comment by JoshuaLuckers** [comment]
I can't even select another Media Source for a Template Variable..

**Comment by Ruslan-Aleev** [comment]
@Mark-H @opengeek @JoshuaLuckers @bezumkin How is it possible that some users have no problems at all, and some have bugs?

**Comment by Ibochkarev** [comment]
Fixed in #14720 
@modxbot close
